indexFormat/specification.md

```diff
@@ -1044,6 +1044,139 @@ export interface MetaData {
+ *
+ * The exact structure of the documentation depends on what makes sense for the specific language
+ * and concepts being described.
+ */
+export interface Documentation extends V {
+	/**
+	 * The label property.
+	 */
+	label: VertexLabels.documentation;
+
+	/**
+	 * A human-readable URL slug identifier for this documentation. It should be unique relative to
+	 * sibling Documentation.
+	 */
+	slug: string;
+
+	/**
+	 * Whether or not this Documentation is the beginning of a new major section, meaning it and its
+	 * its children should be e.g. displayed on their own dedicate page.
+	 */
+	newPage: boolean;
+
+	/**
+	 * A single-line label to display for this documentation in e.g. the index of a book. For
+	 * example, the name of a group of documentation, the name of a library, the signature of a
+	 * function or class, etc.
+	 */
+	title: InteractiveMarkupContent;
+
+	/**
+	 * A detailed multi-line string that contains detailed documentation for the section described by
+	 * the title.
+	 */
+	detail: InteractiveMarkupContent;
+
+	/**
+	 * Tags about the type of content this documentation contains.
+	 */
+	tags: DocumentationTag[];
+
+	/**
+	 * Documentation that should be logically nested below this documentation itself, expressed as
+	 * "documentation" vertex IDs. For example, this documentation may describe a class and have
+	 * children describing each method of the class.
+	 */
+	children: number[];
+}
+```
+Where `InteractiveMarkupContent` is defined as follows:
+```ts
+export interface InteractiveMarkupContent {
+	/**
+	 * The actual content which should be considered interactive.
+	 */
+	content: MarkupContent;
+
+	/**
+	 * Ranges in the `content` string mapping to an associated "documentation" vertex ID, allowing
+	 * for text in one piece of documentation to link to another.
+	 */
+	documentation: { [{range: Range]: number };
+
+	/**
+	 * Ranges in the `content` string mapping to an associated "hoverResult" vertex ID, allowing
+	 * for text in one piece of documentation to include a hover result.
+	 *
+	 * The hover result could be specific to the documentation itself, or e.g. part of a type
+	 * signature being hovered over.
+	 */
+	hover: { [{range: Range]: number };
+
+	/**
+	 * Ranges in the `content` string mapping to an associated "definitionResult" vertex ID,
+	 * allowing for text in one piece of documentation to include a definition result.
+	 *
+	 * This enables users to e.g. go-to-definition on a type signature inside documentation.
+	 */
+	definition: { [{range: Range]: number };
+
+	/**
+	 * Ranges in the `content` string mapping to an associated "referenceResult" vertex ID,
+	 * allowing for text in one piece of documentation to include a definition result.
+	 *
+	 * This enables users to e.g. find-references on a type signature inside documentation.
+	 */
+	references: { [{range: Range]: number };
+}
+```
+Where `DocumentationTag` is defined as follows:
+```ts
+export enum DocumentationTag {

```

Comment:
My thinking here is that a web UI for this could expose check boxes or these tags, so if you check `Unexported` you will see docs for unexported/internal things in packages, same for deprecated etc. Probably the default would be `Exported` but user interfaces themselves can decide that (also based on user preference).

More discussion: https://github.com/sourcegraph/sourcegraph/pull/20108#discussion_r617076494